torchao/quantization/quant_primitives.py

```diff
@@ -888,14 +888,19 @@ def _choose_qparams_affine(
                 "preserve_zero == False is not supported for symmetric quantization"
             )
         if (
-            zero_point_domain is not None
+            zero_point_domain != ZeroPointDomain.NONE.name
+            and zero_point_domain != None
             and zero_point_domain != ZeroPointDomain.INT.name
         ):
             raise ValueError(
-                "zero_point_domain != ZeroPointDomain.INT is not supported for symmetric quantization"
+                "Except a None value for zero_point_domain, Only ZeroPointDomain.NONE and ZeroPointDomain.INT"
+                " are supported for symmetric quantization."
             )
+        if zero_point_domain == ZeroPointDomain.NONE.name:
```

Comment:
thanks for the fix, looks like this is not tested before. can you add a test for the new code path?

also this op is becoming too complicated..we want to split

Reply:
> can you add a test for the new code path?

This case is being tested in a UT I added in `test/quantization/test_quant_primitives.py`

> also this op is becoming too complicated..we want to split

Please advise if you're referring to splitting `_choose_qparams_affine`.
If so, I could split it up into smaller methods. Thanks!

Reply:
yeah I meant splitting choose_qparams_affine/quantize_affine/dequantize to different variations and reduce the complexity of the most common path, this includes removing preserve_zero, zero_point_domain args and just have different variations of choose_qparams_affine/quantize_affine/dequantize. this should be done separately though since it will be a large change